### ISSUE TITLE
Feature request: Allow css or script optimizations on inline CSS or scripts

### ISSUE THREAD

**Issue by ajplopez** [2sic/2sxc#2457]
**I'm submitting a ...**  
[x] feature request

**...about**   
[x] Razor templating

**Current behavior**   
Using the snippets in the code editor, I can load resources like JS or CSS into different parts of the page. This works great for the most part.

However, it seems that we can't use the same methods for moving inline JS or CSS to different parts of the document. Take a look at what I'm trying to do:

![Screen Shot 2021-06-13 at 8 17 34 PM](https://user-images.githubusercontent.com/8751936/121834852-a02d3a00-cc84-11eb-9fe9-741a0c3ee42e.png)

(note that the above code won't move the scripts or css anywhere)

**Expected behavior** 

The inline script or CSS would move to where I'm specifying that it should load.

**Instructions to Reproduce the Problem**
Make inline JS or CSS and try to use the data-enableoptimizations="bottom" on it and notice that it won't actually do anything when the page loads and you view the source - it doesn't move the code.

**Why change the behavior?**  
Occasionally I'll have something like swiper.js where I'm loading the plugin and moving it to the bottom of the page, but I can't move the inline script to the bottom of the page with it. I would either have to leave the plugin script where it is in the module, or make a new file for the inline script and then call it from there.

As for CSS, I don't really use inline CSS at all but I imagine it would be useful to move that to the head of the page in some rare cases.

**Your environment**  
N/A

* **2sxc version(s):** [all]
* **Browser:** [all ]
* **DNN:** [all ] 
* **Language:** [any/all]

**Anything you would like to add**

**Comment by iJungleboy** [comment]
Just for context: data-enableoptimizations is currently meant to integrate _external assets_ with the Dnn Client Dependency Framework to ensure bundeling & proritizing bundles. 

Your example has two distinct things it will try to solve

1. **Activating a JS** once stuff in the DOM is ready - possibly referencing something in the DOM (and often including values like the current `ModuleId` either on `Dnn.Module.ModuleID`, or `CmsContext.Module.Id`. 
2. Providing **inline CSS** overrides

We've been working on the first issue of **activating JS** for a while now in the background, and would like to create a solution which is far more robust and future proof. V12.02.01 actually has a working version included but we're still in the process of testing it to then finalize how it works. We're trying to build it so it solves way more problems - like also pre-ensuring jQuery is loaded, additional dependencies are loaded etc. before even running any `$(function)`. Please check out http://r.2sxc.org/turnon and tell us what you think.

In terms of moving small **inline CSS** down the page I understand what you are doing and assume it's meant for page-speed reasons. ATM I'm not sure if this actually does affect the score, so I'm not sure if it's something that should be built into 2sxc. Or why do you think this is important?

**Comment by ajplopez** [comment]
Regarding the CSS, it's not too important. I don't think it has much, if any, affect on performance, I just thought that moving inline CSS to the head was the standard way of doing things. I personally don't use inline CSS to apply backgrounds via the razor template anymore (I just use Tailwind CSS and position images and use object fit) so this is very, very low priority. We can probably just disregard this one completely, I just thought I would mention it in the off chance it could be easily done when updating the scripts logic.

As for the JS, sounds like #2458 might be the solution.

Let me know if you want me to close this or you can if you want.

**Comment by iJungleboy** [comment]
ok, I'll close this.
